### PR TITLE
Add provision only once information

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Add support for only once provisioned bundles in UI

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/2489

### CODE REVIEW

File: catalog/src/components/ServiceClassList/Cards/styled.js [hunk 1]
```diff
@@ -93,6 +98,67 @@ export const CardHeaderContent = styled(TileContent)`
...
+        font-family: SAP-icons;
+        font-style: normal;
+        font-weight: 400;
+        text-align: center;
+        display: inline-block;
+        text-decoration: inherit;
+        text-transform: none;
+        text-rendering: optimizeLegibility;
+        -webkit-font-smoothing: antialiased;
+        -moz-osx-font-smoothing: grayscale;
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassList/Cards/styled.js]: I know that is a copy from zepplin app, but styled components automatically add css prefixes for appropriated fields. Please remove all `-webkit` etc prefixes.

File: catalog/src/components/ServiceClassList/Cards/styled.js [hunk 2]
```diff
@@ -93,6 +98,67 @@ export const CardHeaderContent = styled(TileContent)`
...
+  && {
+    padding: 2px 6px;
+    background-color: ${props =>
+      props.active === 'true' ? '#0a6ed1' : '#eeeeef'};
+    color: ${props => (props.active === 'true' ? '#ffffff' : '#515559')};
+    border-radius: 2px;
+    font-size: 12px;
+  }
+`;
+export const CardIndicatorProvisionOnce = styled(CardIndicatorGeneral)`
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassList/Cards/styled.js]: if you create extended styled component from `CardIndicatorGeneral` with styles for `provisionOnce ` prop, maybe add those styles to parent component with `provisionOnce` statement?

File: catalog/src/components/ServiceClassList/Cards/Card.component.js [hunk 3]
```diff
@@ -53,6 +77,45 @@ const Card = ({ title, company, description, imageUrl, labels, onClick }) => {
...
+            <CardIndicator>
+              {isProvisionOnlyOnce && (
+                <Tooltip
+                  content={
+                    numberOfInstances > 0
+                      ? tooltipDescription.provisionOnlyOnceActive
+                      : tooltipDescription.provisionOnlyOnce
+                  }
+                >
+                  <CardIndicatorProvisionOnce
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassList/Cards/Card.component.js]: Please use `CardIndicatorGeneral` component with `provisionOnce` prop.

File: catalog/src/components/ServiceClassList/Cards/Card.component.js [hunk 4]
```diff
@@ -33,10 +44,23 @@ const Card = ({ title, company, description, imageUrl, labels, onClick }) => {
       'This Service Class provisions physical resources inside the cluster.',
     showcase:
       'This Service Class demonstrate a specific functionality. Do not use it on the production.',
+  };
+
+  const tooltipDescription = {
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassList/Cards/Card.component.js]: We have to think about extract data like this (description for tolltip) to separate files for easier editing content by TWs, but at the moment this is okay 😸

File: catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js [hunk 5]
```diff
@@ -35,11 +48,42 @@ class ServiceClassDetails extends React.Component {
...
+
+    const buttonText = {
+      provisionOnlyOnce: 'Add once',
+      provisionOnlyOnceActive: 'Added once',
+      standard: 'Add',
+    };
+    const noClassText = "Service Class doesn't exist in this namespace";
+
+    const getModalOpeningComponent = () => {
+      if (isProvisionOnlyOnce && serviceClass && serviceClass.activated) {
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js]: When `isProvisionOnlyOnce` is true, then serviceClass exists, so second condition is always true, so please change those conditions to `isProvisionOnlyOnce && serviceClass.activated`.

File: catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js [hunk 6]
```diff
@@ -35,11 +48,42 @@ class ServiceClassDetails extends React.Component {
...
+        return (
+          <Button option="emphasized" data-e2e-id="add-to-env" disabled={true}>
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js]:
``` jsx
<Button option="emphasized" data-e2e-id="add-to-env" disabled={Boolean(isProvisionOnlyOnce && serviceClass.activated)}>
  {isProvisionOnlyOnce ? (serviceClass.activated ? buttonText.provisionOnlyOnceActive : buttonText.provisionOnlyOnc) : buttonText.standar}
</Button>
```

File: catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js (hunk 5 above)
Comment (magicmatatjahu) [reply catalog/src/components/ServiceClassDetails/ServiceClassDetails.component.js]: Or change whole code for this function to below.

File: catalog/src/components/ServiceClassDetails/ProvisionOnlyOnceInfo/ProvisionOnlyOnceInfo.component.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <ProvisionOnlyOnceInfoContentWrapper
+      colSpan={1}
+      color="#0b74de"
+      data-e2e-id="provision-only-once-info"
+    >
+      <ContentHeader>
+        {noteText.title}
+        <PanelActions>
+          <Icon glyph="message-information" style={{ color: '#0b74de' }} />
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassDetails/ProvisionOnlyOnceInfo/ProvisionOnlyOnceInfo.component.js]: Please override Icon component by creating styled component with this color style.

File: catalog/src/components/ServiceClassDetails/ServiceClassDescription/ServiceClassDescription.component.js [hunk 8]
```diff
@@ -10,9 +10,13 @@ import {
 } from './styled';
 
 const ServiceClassDescription = ({ description }) => {
+  const headerText = 'General Information';
```
Comment (magicmatatjahu) [inline catalog/src/components/ServiceClassDetails/ServiceClassDescription/ServiceClassDescription.component.js]: Very nice 👍

File: catalog/src/components/ServiceClassList/Cards/Card.component.js [hunk 9]
```diff
@@ -33,10 +43,23 @@ const Card = ({ title, company, description, imageUrl, labels, onClick }) => {
       'This Service Class provisions physical resources inside the cluster.',
     showcase:
       'This Service Class demonstrate a specific functionality. Do not use it on the production.',
```
Comment (kazydek) [inline catalog/src/components/ServiceClassList/Cards/Card.component.js]:
```suggestion
      'This Service Class demonstrates a specific functionality. Do not use it on the production.',
```

File: catalog/src/components/ServiceClassList/Cards/Card.component.js [hunk 10]
```diff
@@ -33,10 +43,23 @@ const Card = ({ title, company, description, imageUrl, labels, onClick }) => {
...
+  };
+
+  const tooltipDescription = {
     provisionOnlyOnce:
       'This Service Class can be provisioned only once in a given namespace.',
```
Comment (kazydek) [inline catalog/src/components/ServiceClassList/Cards/Card.component.js]:
```suggestion
      'You can provision this Service Class only once in a given Namespace.',
```

File: catalog/src/components/ServiceClassDetails/ProvisionOnlyOnceInfo/styled.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    color: rgb(50, 54, 58);
+    box-sizing: border-box;
+    font-size: 16px;
+    font-weight: normal;
+  }
+`;
+
+export const ContentDescription = styled(PanelBody)``;
+
+export const Element = styled.div`
```
Comment (aerfio) [inline catalog/src/components/ServiceClassDetails/ProvisionOnlyOnceInfo/styled.js]: Name `Element` feels too generic, maybe something more descriptive would be more  appropriate?

File: catalog/src/components/ServiceClassList/Cards/Card.component.js [hunk 12]
```diff
@@ -33,10 +43,23 @@ const Card = ({ title, company, description, imageUrl, labels, onClick }) => {
...
+    provisionOnlyOnceActive:
+      'This Service Class can be provisioned only once in a given namespace. In this namespace this is already provisioned.',
+    instancesTooltipInfo: 'This Service Class is provisioned',
+    instancesTooltipSingle: 'time.',
+    instancesTooltipPlural: 'times.',
   };
 
+  const isProvisionOnlyOnce =
```
Comment (aerfio) [inline catalog/src/components/ServiceClassList/Cards/Card.component.js]:
```suggestion
  const isProvisionedOnlyOnce =
```

File: tests/ui-tests/tests/catalog-ui-test.js [hunk 13]
```diff
@@ -254,6 +254,12 @@ describeIf(dex.isStaticUser(), 'Catalog basic tests', () => {
       });
 
       expect(catalogHeader).toContain(catalogExpectedHeader);
+
+      console.log('Confirm number of provisioned instances is showed');
```
Comment (aerfio) [inline tests/ui-tests/tests/catalog-ui-test.js]: Confirm that a correct number of provisioned instances is showed? Could you make it clearer?

**Pull request merged**